sdk/webpubsub/azure-messaging-webpubsub/src/main/java/com/azure/messaging/webpubsub/ApimPolicy.java

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.webpubsub;
+
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.util.CoreUtils;
+import reactor.core.publisher.Mono;
+
+import java.net.URL;
+
+/**
+ * The API Management Policy.
+ */
+public final class ApimPolicy implements HttpPipelinePolicy {
```

Comment:
How about a more generic name, for example, ReverseProxyPolicy? Since this can also applicable to Azure Gateway, or Akamai, or other similar products

Reply:
Agree. I have updated the name to ReverseProxyPolicy. I used ApimPolicy because that is what .NET was using. I'll pass this feedback to the .NET team as well.

Reply:
I am [making the change](https://github.com/Azure/azure-sdk-for-net/pull/23830) in .NET.